pkg/resourcecollector/virtualmachine.go

```diff
@@ -1,12 +1,160 @@
 package resourcecollector
 
 import (
+	"fmt"
 	"strings"
 
+	storkapi "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	kubevirtv1 "kubevirt.io/api/core/v1"
 )
 
+const (
+	// PxBackupObjectType_virtualMachine px-backup ObjectType virtualMachine
+	PxBackupObjectType_virtualMachine = "VirtualMachine"
+	//VMPodSelectorLabel is the kubvirt label key for virt-launcher pod of the VM
+	VMPodSelectorLabel = "vm.kubevirt.io/name"
+	//VMFreezeCmd is the template for VM freeze command
+	VMFreezeCmd = "/usr/bin/virt-freezer --freeze --name %s --namespace %s"
+	//VMUnFreezeCmd is the template for VM unfreeze command
+	VMUnFreezeCmd = "/usr/bin/virt-freezer --unfreeze --name %s --namespace %s"
+)
+
+// IsVirtualMachineRunning returns true if virtualMachine is in running state
+func IsVirtualMachineRunning(vm kubevirtv1.VirtualMachine) bool {
+
+	return vm.Status.PrintableStatus == kubevirtv1.VirtualMachineStatusRunning
+}
+
+// GetVMDataVolumes returns DataVolumes used by the VM
+func GetVMDataVolumes(vm kubevirtv1.VirtualMachine) []string {
+
+	volList := vm.Spec.Template.Spec.Volumes
+	dvList := make([]string, 0)
+	for _, vol := range volList {
+		if vol.VolumeSource.DataVolume != nil {
+			dvList = append(dvList, vol.VolumeSource.DataVolume.Name)
+		}
+	}
+	return dvList
+}
+
+// GetVMPersistentVolumeClaims returns persistentVolumeClaim names used by the VMs
+func GetVMPersistentVolumeClaims(vm kubevirtv1.VirtualMachine) []string {
+
+	volList := vm.Spec.Template.Spec.Volumes
+	PVCList := make([]string, 0)
+	for _, vol := range volList {
+		if vol.VolumeSource.PersistentVolumeClaim != nil {
+			PVCList = append(PVCList, vol.VolumeSource.PersistentVolumeClaim.ClaimName)
+		}
+		// Add DataVolume name to the PVC list
+		if vol.VolumeSource.DataVolume != nil {
```

Comment:
So  the array would have one entry of PVC and DV is it? Any reason we want to have DataVolume list returned but the function intention is to fetch only PVC list?

Reply:
We treat DataVolumes as PVC since each dataVolume is associated with a PVC and that PVC is what we will backup.

Reply:
Wouldn't DV be deleted once the user does the image to PVC migration? Also understanding is we don't have to backup up DV?